### PR TITLE
GKE SQL MI scenario initial draft

### PR DESCRIPTION
**Pull request by dkirby-ms**
Added new GKE SQL MI scenario

Addresses #350

### CODE REVIEW

File: azure_arc_data_jumpstart/gke/mssql_mi/terraform/scripts/ClientTools.ps1 [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+# Downloading Azure Data Studio and azdata CLI
+Write-Host "Downloading Azure Data Studio and azdata CLI"
+Write-Host "`n"
+Invoke-WebRequest "https://azuredatastudio-update.azurewebsites.net/latest/win32-x64-archive/stable" -OutFile "C:\tmp\azuredatastudio.zip" | Out-Null
+Invoke-WebRequest "https://raw.githubusercontent.com/microsoft/azure_arc/gke_sqlmi/azure_arc_data_jumpstart/gke/mssql_mi/terraform/settings.json" -OutFile "C:\tmp\settings.json"
+Invoke-WebRequest "https://aka.ms/azdata-msi" -OutFile "C:\tmp\AZDataCLI.msi" | Out-Null
+Invoke-WebRequest "https://raw.githubusercontent.com/microsoft/azure_arc/gke_sqlmi/azure_arc_data_jumpstart/gke/mssql_mi/terraform/scripts/MSSQL_MI_Cleanup.ps1" -OutFile "C:\tmp\MSSQL_MI_Cleanup.ps1"
+Invoke-WebRequest "https://raw.githubusercontent.com/microsoft/azure_arc/gke_sqlmi/azure_arc_data_jumpstart/gke/mssql_mi/terraform/scripts/MSSQL_MI_Deploy.ps1" -OutFile "C:\tmp\MSSQL_MI_Deploy.ps1"
```
Comment (likamrat) [inline azure_arc_data_jumpstart/gke/mssql_mi/terraform/scripts/ClientTools.ps1:35]:
```suggestion
Invoke-WebRequest "https://azuredatastudio-update.azurewebsites.net/latest/win32-x64-archive/stable" -OutFile "C:\tmp\azuredatastudio.zip" | Out-Null
Invoke-WebRequest "https://raw.githubusercontent.com/microsoft/azure_arc/main/azure_arc_data_jumpstart/gke/mssql_mi/terraform/settings.json" -OutFile "C:\tmp\settings.json"
Invoke-WebRequest "https://aka.ms/azdata-msi" -OutFile "C:\tmp\AZDataCLI.msi" | Out-Null
Invoke-WebRequest "https://raw.githubusercontent.com/microsoft/azure_arc/gke_sqlmi/azure_arc_data_jumpstart/gke/main/terraform/scripts/MSSQL_MI_Cleanup.ps1" -OutFile "C:\tmp\MSSQL_MI_Cleanup.ps1"
Invoke-WebRequest "https://raw.githubusercontent.com/microsoft/azure_arc/gke_sqlmi/azure_arc_data_jumpstart/gke/main/terraform/scripts/MSSQL_MI_Deploy.ps1" -OutFile "C:\tmp\MSSQL_MI_Deploy.ps1"
```

File: azure_arc_data_jumpstart/gke/mssql_mi/terraform/variables.tf [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+variable "SPN_TENANT_ID" {
+  description = "Your Azure tenant ID"
+  type        = string
+}
+
+variable "SPN_AUTHORITY" {
+  description = "The Service Principal authority"
+  type        = string
+}
```
Comment (likamrat) [inline azure_arc_data_jumpstart/gke/mssql_mi/terraform/variables.tf:131]:
```suggestion
variable "SPN_AUTHORITY" {
  description = "The Service Principal authority"
  default    = "https://login.microsoftonline.com"
  type        = string
}
```

File: docs/azure_arc_jumpstart/azure_arc_data/gke/gke_mssql_mi_terraform/_index.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+      * Create Arc Data Controller config file (*control.json*) to setup the use of the Storage Class and Kubernetes LoadBalancer service
+      * Deploy the Arc Data Controller **"Directly Connected" mode** using the *TF_VAR* variables values
+      * Execute a secondary *SQLConnectivity* script which will configure the SQL MI instance and download and install the sample Adventureworks database
+      * Unregister the logon script Windows scheduler task so it will not run after first login
+
+## Deployment
+
+As mentioned, the Terraform plan will deploy a GKE cluster, the Azure Arc Data Controller on that cluster, a SQL Managed Instance with sample database, and a Windows Server 2019 Client GCP compute instance.
+
+* Before running the Terraform plan, edit the below *TF_VAR* values and export it (simply copy/paste it after you finished edit these). An example *TF_VAR* shell script file is located [here](https://github.com/microsoft/azure_arc/blob/main/azure_arc_data_jumpstart/gke/dc_vanilla/terraform/example/TF_VAR_example.sh)
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_data/gke/gke_mssql_mi_terraform/_index.md:177]: Looks like the wrong path. Should be the GKE/SQLMI path

**Review by likamrat — CHANGES_REQUESTED** [review]
- 16.png - Take a screenshot without the _tfstate_ and lock files to show the raw folder state before deployment. 
- 34/38.png - Align the controller IP so it will be the same in both screenshots
- Few path updates

File: azure_arc_data_jumpstart/gke/mssql_mi/terraform/scripts/ClientTools.ps1 (hunk 1 above)
Comment (dkirby-ms) [reply azure_arc_data_jumpstart/gke/mssql_mi/terraform/scripts/ClientTools.ps1:35]: This isnt 100% correct and would result in broken links. I will change this after merge.

**Comment by dkirby-ms** [comment]
Comments addressed

**Review by likamrat — APPROVED** [review]
LGTM. Nice work!

**Pull request merged**